### DISCUSSION TITLE
[Question]

### DISCUSSION THREAD

**Discussion by molihuan** [flet-dev/flet#2150] [Q&A]
### Question

I want to obtain configuration data through “self.page.client_storage.get("GroupMsg")” when loading UserControl, but I keep getting errors: 'NoneType' object has no attribute 'client_storage'

### Code sample

```python
class QuickFacebookAccountPage(UserControl):
    def __init__(self, parent: Page):
        super().__init__()
        self.parent = parent

    def initData(self):
        tt = self.page.client_storage.get("GroupMsg")
        print(tt)

    def build(self):
        self.initData()
        return Text("666")
```

### Error message

```bash
AttributeError: 'NoneType' object has no attribute 'client_storage'
```

### ------------------------------------------------------

- [ ] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
Because `self.page` is available after the control is added to a page:

```python
class QuickFacebookAccountPage(UserControl):
    def __init__(self, parent: Page):
        super().__init__()
        self.parent = parent

    def did_mount(self):
        self.initData()

    def initData(self):
        tt = self.page.client_storage.get("GroupMsg")
        print(tt)

    def build(self):
        return Text("666")
```

**Reply by molihuan** [reply]
What should i do.Thanks

**Reply by FeodorFitsner** [reply]
Use `did_mount()`.

**Reply by molihuan** [reply]
> Use `did_mount()`.

i see Thank you very much
